app/models/course/forum.rb

```diff
@@ -6,6 +6,48 @@ class Course::Forum < ActiveRecord::Base
   has_many :topics, dependent: :destroy, inverse_of: :forum
   has_many :subscriptions, dependent: :destroy, inverse_of: :forum
 
+  # @!method self.with_topic_count
+  #   Augments all returned records with the number of topics in that forum.
+  scope :with_topic_count, (lambda do
+    joins { topics.outer }.
+      select { count(topics.id).as(topic_count) }.
+      group { course_forums.id }
+  end)
+
+  # @!method self.with_topic_post_count
+  #   Augments all returned records with the number of topic posts in that forum.
+  scope :with_topic_post_count, (lambda do
+    joins { topics.outer }.
+      joins("LEFT OUTER JOIN course_discussion_topics cdt on
```

Comment:
Problem is that I need to join `course_discussion_topics` table in order to get the actual topic_ids which are referenced by the posts. But with squeel, I can not get this table from the association `act_as :topic` in `Course::Forum::Topic` . So that's why I have to write this normal SQL...